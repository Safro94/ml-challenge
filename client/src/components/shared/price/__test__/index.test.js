import React from 'react';

import { render } from '@testing-library/react';

import formatCurrency from 'utils/currency';

import Price from '../';

jest.mock('utils/currency', () => jest.fn(() => '300,50'));

describe('Price', () => {
	const price = {
		currency: '$',
		amount: 300,
		decimals: 50,
	};

	it('should render OK', () => {
		//Act
		const { container } = render(<Price price={price} />);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should call formatCurrency', () => {
		//Act
		render(<Price price={price} />);

		//Assert
		expect(formatCurrency).toHaveBeenCalledTimes(1);
		expect(formatCurrency).toHaveBeenCalledWith({
			amount: price.amount,
			decimals: price.decimals,
		});
	});
});
