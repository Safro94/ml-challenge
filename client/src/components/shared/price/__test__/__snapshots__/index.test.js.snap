// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should render OK 1`] = `
<div>
  <h1
    class="price"
  >
    <span
      class="currencyCode"
    >
      $
    </span>
    <span>
      300
    </span>
    <span
      class="decimals"
    >
      50
    </span>
  </h1>
</div>
`;
